patterns/callback/callbacker/src/main.nr

```diff
@@ -0,0 +1,128 @@
+mod types;
+mod interfaces;
+
+/// @title Callbacker Contract
+/// @notice This contract handles the callback logic for both atomic and async callbacks.
+contract Callbacker {
+
+    use crate::interfaces::CallMeBack::{CallMeBackPrivateContextInterface, OriginalCallerCallbackStruct, ResolverNewAsyncCallStruct, AllowedAddressNewAsyncCallStruct};
+    use crate::types::pending_callback_note::{PendingCallbackNote, PendingCallbackNoteMethods, PENDING_CALLBACK_NOTE_LEN};
+
+    use dep::aztec::{
+        context::Context,
+        note::note_getter_options::NoteGetterOptions,
+        protocol_types::address::AztecAddress,
+        state_vars::set::Set,
+        log::emit_encrypted_log,
+        oracle::{
+            nullifier_key::get_nullifier_secret_key,
+            get_public_key::get_public_key,
+        },
+    };
+
+    use dep::aztec::note::{
+        note_header::NoteHeader,
+        note_interface::NoteInterface,
+        note_viewer_options::NoteViewerOptions,
+        utils as note_utils,
+
+    };
+
+    use dep::std::option::Option;
+    
+    global PENDING_CALLBACK_NOTE_SLOT = 1;
+
+    /// @notice Storage struct for pending callback notes
```

Comment:
I would add a small explanation like we have with the `call_me_back` to make it very clear what function is related to async and atomic